src/averageAges.js

```diff
@@ -55,14 +78,25 @@
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const children = people.filter((person, _, arr) => {
+    const hasMother = arr.find(mother => mother.name === person.mother);
+
+    return hasMother && (onlyWithSon ? person.sex === 'm' : true);
+  });
+
+  const childrensTotalAge = children.reduce((acc, child,) => {
+    const mother = people.find(person => person.name === child.mother);
+
+    if (mother !== child.born) {
+      return acc + child.born - mother.born;
+    }
+  }, 0);
+
+  return childrensTotalAge / children.length;
```

Comment:
The calculation of `childrensTotalAge` seems to have a logical error. Check if `mother` is not `undefined` before comparing it to `child.born`. Also, ensure that you're calculating the age difference correctly.